src/Uno.Extensions.Validation/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Uno.Extensions.Validation;
+
+public static class ServiceCollectionExtensions
+{
+	public static IServiceCollection AddInstanceTypeInfo<TEntity>(
+	this IServiceCollection services
+	)where TEntity : class
+	{
+		return services
+			.AddSingleton<IInstanceType>(sp => new IInstanceTypeWrapper<TEntity>(sp));
+		
+	}
+}
+
+internal interface IInstanceType : IValidator
```

Comment:
I think we should rename this to IValidatorTypedInstance (and we should do the same for the serialization IInstanceType -> ISerializerTypedInstance)